classes/class.ilMumieTaskGradeSync.php

```diff
@@ -205,4 +227,49 @@ private function getLatestGrade($xapi_grades)
         }
         return $latest_grade;
     }
+
+    public static function checkIfGradeWasAchievedByUser($user_id, $parentObj, $grade)
+    {
+        $user_grades = self::getGradesForUser($user_id, $parentObj);
+        foreach ($user_grades as $xapi_grade) {
+            if ($grade == round($xapi_grade->result->score->raw * 100)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public static function getGradesForUser($user_id, $parentObj)
+    {
+        $gradesync  = new  ilMumieTaskGradeSync($parentObj->object, false);
+        $xapi_grades = $gradesync->getAllXapiGradesByUser();
+        $syncId = $gradesync->getSyncIdForUser($user_id);
+        $userGrades = array();
+        if (empty($xapi_grades)) {
+            return;
+        }
+        foreach ($xapi_grades as $xapi_grade) {
+            if ($xapi_grade->actor->account->name == $syncId) {
+                array_push($userGrades, $xapi_grade);
+            }
+        }
+        return $userGrades;
+    }
+
+    public static function getFirstName($user_id)
+    {
+        return self::getUsername($user_id)["firstname"];
+    }
+
+    public static function getLastName($user_id)
+    {
+        return self::getUsername($user_id)["lastname"];
+    }
+
+    private static function getUsername($user_id)
+    {
+        global $ilDB;
+        $result = $ilDB->query("SELECT firstname, lastname FROM usr_data WHERE usr_id = ". $ilDB->quote($user_id, "integer"));
+        return $ilDB->fetchAssoc($result);
+    }
```

Comment:
These methods should not be part of this class as they are not related to grade synchronization. Instead, create a new class ilMumieTaskUserServer

Reply:
done

Reply:
I misspelled my suggested class name. It should be `ilMumieTaskUserService` not `Server`. Sorry

Reply:
Fixed it